miner/worker.go

```diff
@@ -451,6 +453,20 @@ func (w *worker) mainLoop() {
 					w.commit(uncles, nil, true, start)
 				}
 			}
+		// Read from the sync service and mine single txs
+		// as they come. Wait for the block to be mined before
+		// reading the next tx from the channel when there is
+		// not an error processing the transaction.
+		case ev := <-w.rollupCh:
+			tx := ev.Txs[0]
+			if err := w.commitNewTx(nil, tx); err == nil {
+				head := <-w.chainHeadCh
+				txn := head.Block.Transactions()[0]
```

Comment:
This could use a safety check on `len(head.Block.Transactions())`

Reply:
Yeah this will panic on empty blocks, checking that len is greater than 0 should be enough